torch/_dynamo/convert_frame.py

```diff
@@ -165,19 +167,24 @@ def _fn(*args, **kwargs):
             return fn(*args, **kwargs)
         finally:
             cleanup.close()
-            torch._C._set_grad_enabled(prior_grad_mode)
-            torch.torch.autograd.grad_mode._enter_inference_mode(prior_inference_mode)
-            torch.use_deterministic_algorithms(
-                prior_deterministic, warn_only=prior_warn_only
+            should_pop_torch_mode = (_pop_mode_temporarily()
```

Comment:
Is it True that if torch._C._is_torch_function_mode_enabled(), we can always pop the top of the stack?

Reply:
Maybe i will have to go through all torch function modes in the stack and only disable PreDispatchTOrchFunctionMode? But would that make torch.compile slow?